provider/pkg/default_tags/merging.go

```diff
@@ -33,6 +33,12 @@ func MergeDefaultTags(tags resource.PropertyValue, defaultTags map[string]string
 
 	if tagsStyle.IsKeyValueArray() {
 		var asArray []resource.PropertyValue
+		keyProp := resource.PropertyKey("key")
+		valueProp := resource.PropertyKey("value")
+		if tagsStyle == TagsStyleKeyValueArrayUpperCase {
+			keyProp = resource.PropertyKey("Key")
+			valueProp = resource.PropertyKey("Value")
+		}
```

Comment:
hrmmm... this delta is a bummer since it seems to just be a matter of skipping name normalization for the extension resource. I suppose normalizing names would be an even more messy change though.  Would like to see a unit test for this though.